cheddar/cheddar-persistence/src/main/java/com/clicktravel/cheddar/infrastructure/persistence/document/search/query/AndQuery.java

```diff
@@ -40,4 +40,35 @@ public void addQuery(final StructuredQuery query) {
     public void accept(final QueryVisitor visitor) {
         visitor.visit(this);
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((queries == null) ? 0 : queries.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final AndQuery other = (AndQuery) obj;
+        if (queries == null) {
+            if (other.queries != null) {
+                return false;
+            }
+        } else if (!queries.equals(other.queries)) {
```

Comment:
When comparing `queries`, the order of the List should not matter i.e. this will fail comparing queries "A AND B" and "B AND A". I think `queries` should be a Set, but this will require testing.